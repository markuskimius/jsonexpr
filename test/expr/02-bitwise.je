PRINT("== OPERATION ==");
PRINT(0xFF00 & 0xF0F0);
PRINT(0xFF00 | 0xF0F0);
PRINT(0xFF00 ^ 0xF0F0);
PRINT(~0xFF00);
PRINT();

PRINT("== PRECEDENCE ==");
PRINT(0xFFFF0000 & 0xFF00FF00 ^ 0xF0F0F0F0);
PRINT(0xFFFF0000 & 0xFF00FF00 | 0xF0F0F0F0);
PRINT(0xFFFF0000 ^ 0xFF00FF00 & 0xF0F0F0F0);
PRINT(0xFFFF0000 ^ 0xFF00FF00 | 0xF0F0F0F0);
PRINT(0xFFFF0000 | 0xFF00FF00 & 0xF0F0F0F0);
PRINT(0xFFFF0000 | 0xFF00FF00 ^ 0xF0F0F0F0);
PRINT();

PRINT(0xFFFF0000 & ~0xFF00FF00 ^ 0xF0F0F0F0);
PRINT(0xFFFF0000 & ~0xFF00FF00 | 0xF0F0F0F0);
PRINT(0xFFFF0000 ^ ~0xFF00FF00 & 0xF0F0F0F0);
PRINT(0xFFFF0000 ^ ~0xFF00FF00 | 0xF0F0F0F0);
PRINT(0xFFFF0000 | ~0xFF00FF00 & 0xF0F0F0F0);
PRINT(0xFFFF0000 | ~0xFF00FF00 ^ 0xF0F0F0F0);
PRINT();
