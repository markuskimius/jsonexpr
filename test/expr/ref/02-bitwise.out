== OPERATION ==
61440
65520
4080
-65281

== PRECEDENCE ==
267448560
4293980400
268431360
4043309040
4294963200
4294905840

4027576560
4043305200
4279173360
-986881
4294902000
-4081

